tools/monitoring/mysql/alert.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mysql
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/knative/test-infra/tools/monitoring/config"
+	"github.com/knative/test-infra/tools/monitoring/mail"
+)
+
+const (
+	alertInsertStmt = `
+		INSERT INTO Alerts (
+			Sent, ErrorPattern
+		) VALUES (?,?)`
+
+	alertUpdateStmt = `
+	UPDATE Alerts
+	SET Sent = (?)
+	WHERE ErrorPattern = (?)`
+
+	emailTemplate = `In the past %v, 
```

Comment:
For the email template, it's missing a few fields. From the design, these are the fields that'll be included: 

* Error message pattern
* Error message
* Number of occurrences
* Jobs affected along with a link to the gcs artifact directory that contains the build logs
* Pull Requests affected
* Recovery suggestions. 

You can try using [text/template](https://golang.org/pkg/text/template/) to generate the error message if the format string gets too long.

Reply:
The missing fields are already mentioned in the PR description. This PR is already complicated and we should do that in a separate PR, with some function rewritten to get the needed fields

I'll look into text template